mlir/lib/Dialect/Affine/Analysis/LoopAnalysis.cpp

```diff
@@ -344,10 +333,12 @@ bool mlir::affine::isVectorizableLoopBody(
     auto load = dyn_cast<AffineLoadOp>(op);
     auto store = dyn_cast<AffineStoreOp>(op);
     int thisOpMemRefDim = -1;
-    bool isContiguous = load ? isContiguousAccess(loop.getInductionVar(), load,
-                                                  &thisOpMemRefDim)
-                             : isContiguousAccess(loop.getInductionVar(), store,
-                                                  &thisOpMemRefDim);
+    bool isContiguous =
+        load ? isContiguousAccess(loop.getInductionVar(),
+                                  (AffineReadOpInterface)load, &thisOpMemRefDim)
```

Comment:
Nit: I'd rather `cast<...>(load)`, C-style casting presumably works thanks to interface constructor from `Operation *` being public, but it isn't a good practice to use that.

Reply:
But `cast<interface>(load)` won't work here - note that `load` is a derived op type!
```
...lib/Dialect/Affine/Analysis/LoopAnalysis.cpp:338:62:   required from here
.../llvm/include/llvm/Support/Casting.h:64:64: error: cannot convert ‘const mlir::affine::AffineLoadOp*’ to ‘mlir::Operation*’
   64 |   static inline bool doit(const From &Val) { return To::classof(&Val); }
```
You'll have to do `cast<interface>(*load)`, which would work. Would you prefer this still?

Reply:
Yes, please. 

I think I saw a discussion about making the pointer-based constructor private, but I don't recall where.

Reply:
Sure, done.